packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -916,45 +916,39 @@ async function getTagAndPage(componentTestSetup: ComponentTestSetup): Promise<Ta
 /**
  * Helper to test the disabled prop disabling user interaction.
  *
- * @param {ComponentTestSetup} componentTestSetup - A component tag, html, or the tag and e2e page for setting up a test
- * @param {DisabledOptions} [options={ focusTarget: "host" }] - disabled options
+ * Note that this helper should be used within a describe block.
+ *
+ * @example
+ * describe("disabled", () => {
+ *    disabled("calcite-input")
+ * });
+ *
+ * @param {ComponentTestSetup} componentTestSetup - A component tag, html, or the tag and e2e page for setting up a test.
+ * @param {DisabledOptions} [options={ focusTarget: "host" }] - Disabled options.
  */
-export async function disabled(
+export function disabled(
   componentTestSetup: ComponentTestSetup,
   options: DisabledOptions = { focusTarget: "host" }
-): Promise<void> {
-  const { page, tag } = await getTagAndPage(componentTestSetup);
-
-  const component = await page.find(tag);
-  await skipAnimations(page);
-  await page.$eval(tag, (el) => {
-    el.addEventListener(
-      "click",
-      (event) => {
-        const path = event.composedPath() as HTMLElement[];
-        const anchor = path.find((el) => el?.tagName === "A");
-
-        if (anchor) {
-          // we prevent the default behavior to avoid a page redirect
-          anchor.addEventListener("click", (event) => event.preventDefault(), { once: true });
-        }
-      },
-      true
-    );
-  });
-
-  // only testing events from https://github.com/web-platform-tests/wpt/blob/master/html/semantics/disabled-elements/event-propagate-disabled.tentative.html#L66
-  const eventsExpectedToBubble = ["mousemove", "pointermove", "pointerdown", "pointerup"];
-  const eventsExpectedToNotBubble = ["mousedown", "mouseup", "click"];
-  const allExpectedEvents = [...eventsExpectedToBubble, ...eventsExpectedToNotBubble];
-
-  const eventSpies: EventSpy[] = [];
-
-  for (const event of allExpectedEvents) {
-    eventSpies.push(await component.spyOnEvent(event));
-  }
+): void {
+  const addClickListenersWithRedirectPrevention = async (page: E2EPage, tag: string): Promise<void> => {
```

Comment:
I think we can drop the click listeners part since it's an implementation detail. The redirect prevention is the key part. `addClickListenersWithRedirectPrevention` ➡️ `addRedirectPrevention`/`preventRedirects`?